arm/Microsoft.Resources/resourceGroups/deploy.bicep

```
@@ -43,13 +43,13 @@ resource resourceGroup 'Microsoft.Resources/resourceGroups@2019-05-01' = {
   properties: {}
 }
 
-module resourceGroup_lock '.bicep/nested_lock.bicep' = if (lock != 'NotSpecified') {
-  scope: resourceGroup
+module resourceGroup_lock '.bicep/nested_lock.bicep' = if (!empty(lock)) {
   name: '${uniqueString(deployment().name, location)}-RG-${lock}-Lock'
```

Aligning with the rest of the names. 

```suggestion
  name: '${uniqueString(deployment().name, location)}-${lock}-Lock'
```